nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -519,18 +519,58 @@ def build_train_valid_test_datasets(self):
             eval_iters * global_batch_size,
             test_iters * global_batch_size,
         ]
-        self._train_ds, self._validation_ds, self._test_ds = build_train_valid_test_datasets(
-            cfg=self.cfg,
-            trainer=self.trainer,
-            data_prefix=self.cfg.data.data_prefix,
-            data_impl=self.cfg.data.data_impl,
-            splits_string=self.cfg.data.splits_string,
-            train_valid_test_num_samples=train_valid_test_num_samples,
-            seq_length=self.cfg.data.seq_length,
-            seed=self.cfg.seed,
-            skip_warmup=self.cfg.data.get('skip_warmup', True),
-            tokenizer=self.tokenizer,
-        )
+        if isinstance(self.cfg.data.data_prefix, DictConfig):
+            assert self.cfg.data.data_prefix.get('train') is not None and self.cfg.data.data_prefix.get('test') is not None and self.cfg.data.data_prefix.get('validation') is not None, f"Data prefix dictionary should have train, test and validation keys.  data_prefix currently has only {self.cfg.data.data_prefix.keys()}"
+            logging.warning(self.cfg.data.splits_string  + "ignored since data prefix is of type dictionary.")
```

Comment:
Maybe log the warning here only if splits string is not None?

Reply:
Sure makes sense.